### PR TITLE
[scripts] Make generate_plots.py python3-compatible

### PR DESCRIPTION
**Pull request by kkm000**
* Use raw strings for TeX; python3 has more escapes.
 * Use sorted() for sorting, since many functions return
   transient iterators.

Also, mostly cosmetic fixes:
 * Do not print the missing imports warning before parsing
   the command line (it can use --help).
 * Suppress false-positive matplotlib warning.
 * Add metavars to help message so it reads better.
 * Make logging lines more concise (no point printing time).
 * Simplify logger initialization.
 * Use logging.warning for warnings; warnings.warn() is for
   library code only.

### CODE REVIEW

**Comment by vijayaditya** [comment]
Thanks will take a look in 3 hours.

**Comment by danpovey** [comment]
Thanks, Vijay!

File: egs/wsj/s5/steps/nnet3/report/generate_plots.py [hunk 1]
```diff
@@ -145,8 +132,9 @@ def compile(self):
                 "pdflatex -interaction=batchmode "
                 "-output-directory={0} {1}".format(dir_name, latex_file))
         except Exception as e:
-            logger.warning("There was an error compiling the latex file {0}, "
-                           "please do it manually: {1}".format(latex_file, e))
+            logger.warning("There was an error compiling latex file {0}. "
+                           "Check report.log generated by pdflatex in the same directory. "
+                           "{1}".format(latex_file, e))
```
Comment (vijayaditya) [inline egs/wsj/s5/steps/nnet3/report/generate_plots.py]: Could you modify this string to use "%s" specification similar to the other modifications you made.

File: egs/wsj/s5/steps/nnet3/report/generate_plots.py [hunk 2]
```diff
@@ -23,76 +23,62 @@
...
-                        "e.g. exp/nnet3/tdnn/report")
+                        help="output directory for reports, e.g. 'exp/nnet3/tdnn/report'")
 
     args = parser.parse_args()
-    if (args.comparison_dir is not None and len(args.comparison_dir) > 6) or \
-    (args.exp_dir is not None and len(args.exp_dir) > 7):
+    if ((args.comparison_dir is not None and len(args.comparison_dir) > 6) or
+        (args.exp_dir is not None and len(args.exp_dir) > 7)):
         raise Exception(
             """max 6 comparison directories can be specified.
```
[the remark this answers is not in the document]
Comment (kkm000) [reply egs/wsj/s5/steps/nnet3/report/generate_plots.py]: Done (+2 or 3 other places).

File: egs/wsj/s5/steps/nnet3/report/generate_plots.py [hunk 3]
```diff
@@ -675,9 +650,8 @@ def generate_parameter_diff_plots(exp_dir, output_dir, plot,
         assert main_component_names
 
         fig = plt.figure()
-        logger.info("Generating parameter-difference plots for the "
-                    "following components:{0}".format(
-                        ', '.join(main_component_names)))
+        logger.info("Plotting parameter differences for components: " +
+                    ', '.join(main_component_names))
```
[the remark this answers is not in the document]
Comment (kkm000) [reply egs/wsj/s5/steps/nnet3/report/generate_plots.py]: Done.

File: egs/wsj/s5/steps/nnet3/report/generate_plots.py (hunk 1 above)
Comment (kkm000) [reply egs/wsj/s5/steps/nnet3/report/generate_plots.py]: Done, in all logger.xxx invocations, I believe.

**Comment by vijayaditya** [comment]
Thanks a lot. I will wait for @danpovey to make the merge.

**Comment by kkm000** [comment]
Thanks for the review!

**Pull request merged**